### PR TITLE
cleanup: move the test of TaintBasedEvictions features to sig-node

### LINKED ISSUE #87427: [TaintBasedEvictions] Ensure test coverage is complete and up-to-date
As part of promoting TaintBasedEvictions to GA in 1.18 (https://github.com/kubernetes/kubernetes/issues/87161), we need to make sure that the test coverage is sufficient. There are some tests mentioned in the KEP, and as mentioned in https://github.com/kubernetes/enhancements/pull/1450#discussion_r366899543, those that are still under scheduling should be moved to the appropriate SIGs (sig-node or sig-cloud-provider)

/kind cleanup

### PR DESCRIPTION
**Pull request by fengzixu**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

1. move the integration test of TaintBasedEvictions to test/integration/node
2. move the e2e test of TaintBasedEvictions to e2e_node

**Which issue(s) this PR fixes**:

Fixes #87427

**Special notes for your reviewer**:
```release-note
NONE
```

### CODE REVIEW

**Comment by fengzixu** [comment]
@Huang-Wei @damemi 

Hi guys. I finished the moving tests from sig-scheduling to sig-node based on #87427.

Can you help me review it and ping members of sig-node. Thank you so much.

**Comment by damemi** [comment]
cc @kubernetes/sig-node-pr-reviews

File: test/e2e_node/BUILD
Comment (mattjmcnaughton) [inline test/e2e_node/BUILD]: I think you're missing a comma here :) which is I think responsible for pretty much all of the test failures :)

File: test/integration/node/OWNERS
Comment (mattjmcnaughton) [inline test/integration/node/OWNERS]: Nit: Why are we not specifying approvers?

File: test/integration/node/util.go
Comment (mattjmcnaughton) [inline test/integration/node/util.go]: Perhaps I'm missing something, but where is this `util.go` code coming from? I don't see the equivalent code removed anywhere. Are we duplicating it?

**Review by mattjmcnaughton — COMMENTED** [review]
Thanks for this pr :)

Agree with the high-level goal. I have a couple of impl qs :)

File: test/e2e_node/BUILD
Comment (damemi) [reply test/e2e_node/BUILD]: yeah, not sure if this was done manually but if so running `./hack/update-bazel.sh` is how these should be updated

File: test/integration/node/OWNERS
Comment (fengzixu) [reply test/integration/node/OWNERS]: emm. Actually, I don't know who I should specify. sig-node or sig-scheduling. Can you give me hint?

File: test/e2e_node/BUILD
Comment (fengzixu) [reply test/e2e_node/BUILD]: Is BUILD file can be generated automatically? Any script can do it?

File: test/integration/node/util.go
Comment (fengzixu) [reply test/integration/node/util.go]: Actually, some functions which are used by the integration test of TaintBasedEvictions exists in the `test/integration/scheduling/util.go`. In my opinion, if we move the TaintBasedEviction test to the sig-node, we shouldn't import unnecessary functions from sig-scheduling.

File: test/integration/node/OWNERS
Comment (fengzixu) [reply test/integration/node/OWNERS]: Let me fill it.

Comment (fengzixu) [reply test/integration/node/OWNERS]: Who I need to speficy? It seems like sig-node-maintainers is incorrect.

File: test/e2e_node/BUILD
Comment (mattjmcnaughton) [reply test/e2e_node/BUILD]: Yup, `hack/update-bazel.sh` is the script which will do it.

File: test/integration/node/util.go
Comment (mattjmcnaughton) [reply test/integration/node/util.go]: Hmm, I definitely agree that we don't want to import from `test/integration/scheduling/util.go`. However, I'd prefer to avoid the duplication if we can. Can we move `util.go` to a more general package which both `scheduling` and `node` could safetly import from?

Alternatively, if there's a strict timeline for getting this merged, we could also address this duplication in a later pr. If we go that route, could you please add a TODO to resolve the duplication?

File: test/integration/node/OWNERS
Comment (mattjmcnaughton) [reply test/integration/node/OWNERS]: `sig-node-approvers` is the proper alias.

File: test/e2e_node/taints.go
Comment (Huang-Wei) [inline test/e2e_node/taints.go]: `test/e2e_node` uses a totally different e2e mechanism. We should move it to `test/e2e/node`.

File: test/integration/node/OWNERS
Comment (Huang-Wei) [inline test/integration/node/OWNERS]:
```suggestion
- sig-node-approvers
```

File: test/integration/node/util.go
Comment (Huang-Wei) [reply test/integration/node/util.go]: `test/integration/util` would be a good place.

**Review by Huang-Wei — COMMENTED** [review]
Some comments.

File: test/integration/node/util.go
Comment (fengzixu) [reply test/integration/node/util.go]: Yep. Let me move it .

**Comment by fengzixu** [comment]
@manansaraf @Huang-Wei 

Hey guys. I revised this PR according to the comments above.  Please review it again.

**Review by mattjmcnaughton — COMMENTED** [review]
Mind addressing the failures? https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/88152/pull-kubernetes-verify/1229445261944688641

Lmk if there any questions around how to address the failures - you can verify if they are working via the `hack/verify-*` scripts.

Happy to review if you have specific questions or once the test suite is green. Thanks!

**Comment by fengzixu** [comment]
> Mind addressing the failures? https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/88152/pull-kubernetes-verify/1229445261944688641
> 
> Lmk if there any questions around how to address the failures - you can verify if they are working via the `hack/verify-*` scripts.
> 
> Happy to review if you have specific questions or once the test suite is green. Thanks!

I see. I missed it. Let me revise it.

**Comment by fengzixu** [comment]
@mattjmcnaughton Not sure why the pull-kubernetes-bazel-test  always failed. 
The report of the test doesn't give accurate info: https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/88152/pull-kubernetes-bazel-test/1229637653826113536

**Comment by fengzixu** [comment]
@Huang-Wei @mattjmcnaughton It seems like all tests have passed. So please review it

**Review by mattjmcnaughton — APPROVED** [review]
/lgtm

Thanks for this change and for addressing all the review feedback.

The one thing I'm not sure about, and would like someone to sign-off on, is whether we need to make any other changes similar to the change to the `conformance.txt` test file.

File: test/integration/node/main_test.go
Comment (Huang-Wei) [inline test/integration/node/main_test.go]: nit: 2020

**Comment by Huang-Wei** [comment]
/approve
for the scheduling side. Could you squash the commits?

**Comment by fengzixu** [comment]
@Huang-Wei @smarterclayton @mattjmcnaughton
I have rebased upstream and combine multiple commits to one. Please review it.

/assign @smarterclayton

File: test/integration/node/main_test.go
Comment (fengzixu) [reply test/integration/node/main_test.go]: fixed

**Review by mattjmcnaughton — APPROVED** [review]
/lgtm

Still good on my end if all you did was squash (again, modulo open conformance question).

Will let @smarterclayton give final sign-off.

**Comment by fengzixu** [comment]
conflict happened on /test/e2e/node/BUILD file. So I rebase it.

**Comment by fengzixu** [comment]
Ping @smarterclayton :)

**Review by damemi — APPROVED** [review]
/lgtm
will need an approve though

**Comment by smarterclayton** [comment]
All refactoring of utility methods should be split into its own PR and merged first.  This PR must be two commits - one changing the conformance file, and one moving the files and renaming things.

**Comment by fengzixu** [comment]
> All refactoring of utility methods should be split into its own PR and merged first. This PR must be two commits - one changing the conformance file, and one moving the files and renaming things.

has split, please review. @smarterclayton

**Comment by mattjmcnaughton** [comment]
> > All refactoring of utility methods should be split into its own PR and merged first. This PR must be two commits - one changing the conformance file, and one moving the files and renaming things.
> 
> has split, please review. @smarterclayton

@smarterclayton can confirm, but I think that he was actually suggesting two separate PRs, not one PR containing two commits (which is what I believe we have now). I think we want the ability to first merge the utility methods and then merge the conformance changes.

Do you mind creating two separate PRs so that we can merge the changes independently? If you tag me in both, happy to review tomorrow morning. Thanks :)

**Comment by fengzixu** [comment]
> > > All refactoring of utility methods should be split into its own PR and merged first. This PR must be two commits - one changing the conformance file, and one moving the files and renaming things.
> > 
> > 
> > has split, please review. @smarterclayton
> 
> @smarterclayton can confirm, but I think that he was actually suggesting two separate PRs, not one PR containing two commits (which is what I believe we have now). I think we want the ability to first merge the utility methods and then merge the conformance changes.
> 
> Do you mind creating two separate PRs so that we can merge the changes independently? If you tag me in both, happy to review tomorrow morning. Thanks :)
> 
> Alternatively, if the two separate PRs already exist, and I'm just missing them, do you mind linking them?

@mattjmcnaughton Sure. Let me split it into 2 PRs.

**Comment by fengzixu** [comment]
https://github.com/kubernetes/kubernetes/pull/88685 @smarterclayton @mattjmcnaughton Have split, PTAL

**Comment by mattjmcnaughton** [comment]
I think we will need to rebase after https://github.com/kubernetes/kubernetes/pull/88685 is merged? Mind pinging me once that's done?

Thanks :)

**Comment by damemi** [comment]
@fengzixu @smarterclayton with code freeze this week, will this be able to merge? The other 2 PRs to complete this enhancement (https://github.com/kubernetes/kubernetes/pull/87487 & https://github.com/kubernetes/website/pull/19302) are predicated on this merging first

**Comment by fengzixu** [comment]
PR https://github.com/kubernetes/kubernetes/pull/88685 has been merged last night. I will rebase for this PR this night. After that, please @smarterclayton review and approve it.

@damemi Is it ok?

**Comment by fengzixu** [comment]
@mattjmcnaughton @smarterclayton  Hi, I have rebased it and all test cases seem passed. Please review this PR. Because of the code freeze will start at the end of this week.

**Review by mattjmcnaughton — COMMENTED** [review]
Will let @smarterclayton give the final sign-off since I don't have approval permissions.

/assign @smarterclayton

**Review by damemi — APPROVED** [review]
/lgtm
Thanks for this @fengzixu!

**Comment by damemi** [comment]
Bumping, @smarterclayton will you have time to review this today for code freeze? If not can we file an exception for the enhancement?

**Pull request merged**
